frontend/src/host_orchestrator/orchestrator/instancemanager.go

```diff
@@ -96,22 +97,23 @@ type CVDToolInstanceManagerOpts struct {
 }
 
 func NewCVDToolInstanceManager(opts *CVDToolInstanceManagerOpts) *CVDToolInstanceManager {
+	downloadCVDHandler := newDownloadCVDHandler(
+		opts.CVDBinAB,
+		opts.Paths.CVDBin,
+		opts.BuildAPIFactory(""), // cvd can be downloaded without credentials
+	)
 	return &CVDToolInstanceManager{
 		execContext:              opts.ExecContext,
 		paths:                    opts.Paths,
 		om:                       opts.OperationManager,
 		userArtifactsDirResolver: opts.UserArtifactsDirResolver,
 		hostValidator:            opts.HostValidator,
-		downloadCVDHandler: &downloadCVDHandler{
-			Build:    opts.CVDBinAB,
-			Filename: opts.Paths.CVDBin,
-			BuildAPI: opts.BuildAPIFactory(""), // cvd can be downloaded without credentials
-		},
+		downloadCVDHandler:       downloadCVDHandler,
 		artifactsMngr: NewArtifactsManager(
 			opts.Paths.ArtifactsRootDir,
 			opts.UUIDGen,
 		),
-		buildArtifactsFetcher: newArtifactsFetcherFactory(opts.ExecContext, opts.Paths.CVDBin, opts.BuildAPIFactory),
+		buildArtifactsFetcher: newArtifactsFetcherFactory(opts.ExecContext, downloadCVDHandler.fetchCVDFilename, opts.BuildAPIFactory),
```

Comment:
rather than computing `fetch_cvd` filename in `newDownloadCVDHandler`  based on `opts.Paths.CVDBin` value, I'd suggest to pass a new option `opts.Paths.CVDBinDir` rather than `opts.Paths.CVDBin`, using opts.Paths.CVDBinDir + '/cvd' and opts.Paths.CVDBinDir + '/fetch_cvd' wherever relevant, I think it would be more transparent.

Reply:
Done